src/shims/panic.rs

```diff
@@ -58,7 +54,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         thread.panic_payload = Some(payload);
 
         // Jump to the unwind block to begin unwinding.
-        this.unwind_to_block(unwind);
+        this.unwind_to_block(match unwind {
```

Comment:
This looks like `unwind_to_block` should really take a `StackPopUnwind`.

Reply:
That's a great idea, and that means this `throw_ub_format!` should be moved inside `unwind_to_block()`, and `pop_stack_frame()` calls `unwind_to_block()`, which means it also does not need its own `throw_ub_format!` anymore. However, that also means we can't do https://github.com/rust-lang/rust/pull/85546#discussion_r638036854 anymore, unless we want to duplicate this check before popping the frame, or pushing the frame back before `throw_ub_format!`.

Reply:
Ah, good point... maybe we leave improving the span to the future? FWIW, returning from a function when the return block is `None` has the exact same problem; that error is currently checked in `return_to_block`.